src/sql/workbench/browser/designer/designer.ts

```diff
@@ -904,11 +976,25 @@ export class Designer extends Disposable implements IThemable {
 					table.registerPlugin(deleteRowColumn);
 					columns.push(deleteRowColumn.definition);
 				}
-				table.columns = columns;
+				table.columns = buttonColumn.concat(columns);
+				table.columns.forEach((v, i) => table.columns[i].id = table.columns[i].id || i.toString());
 				table.grid.onBeforeEditCell.subscribe((e, data): boolean => {
 					return data.item[data.column.field].enabled !== false;
 				});
-
+				let disabledActions = this._actions.filter(b => b.enabled === false);
+				table.grid.onClick.subscribe((e, data) => {
+					const dataLength = table.grid.getDataLength();
+					disabledActions.forEach((b) => {
+						if (data.row === dataLength - 1 && b instanceof MoveRowDownAction) {
+							b.enabled = false;
+						} else if (data.row === 0 && b instanceof MoveRowUpAction) {
+							b.enabled = false;
+						} else {
+							b.enabled = true;
+						}
+					});
+					table.grid.setSelectedRows([data.row]);
```

Comment:
please try different scenarios to see whether data.row is always available. for example, click on the column header and handle it properly if not already.